sorc/chgres_cube.fd/surface.F90

```diff
@@ -96,6 +98,9 @@ module surface
  type(esmf_field)                   :: terrain_from_input_grid
                                        ! terrain height interpolated
                                        ! from input grid
+ type(esmf_field)                   :: terrain_from_input_grid_land
+                                       ! terrain height interpolated
+                                       ! from input grid at all land points 
 
```

Comment:
How is terrain_from_input_grid_land used?  It is computed below.  But I don't understand how it is used.

Reply:
We were experiencing issues where isolated land ice points weren't ever able to obtain a proper terrain height, so this array is passed through during the search/replace of land ice points. Since soil temperature adjustment for terrain height is not turned off when points on the target grid have the default terrain height replacement value (default_value = -99999.9), providing data interpolated to "land points" which the input data originally deemed these land ice points to be (since it doesn't have land ice) allows for these points to have a proper terrain height. It looks like this use actually got coded out, but I'm putting it back in by providing the data from terrain_from_input_grid as an optional argument to the search function on line 2049 of surface.F90.